x-pack/auditbeat/module/system/user/user.go

```diff
@@ -0,0 +1,303 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package user
+
+import (
+	"bytes"
+	"encoding/gob"
+	"io"
+	"strconv"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/common/cfgwarn"
+	"github.com/elastic/beats/metricbeat/mb"
+
+	"github.com/OneOfOne/xxhash"
+
+	"github.com/elastic/beats/auditbeat/datastore"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/x-pack/auditbeat/cache"
+)
+
+const (
+	moduleName    = "system"
+	metricsetName = "user"
+	bucketName    = "user.v1"
+
+	eventTypeState = "state"
+	eventTypeEvent = "event"
+
+	eventActionUserExists  = "existing_user"
+	eventActionUserAdded   = "user_added"
+	eventActionUserRemoved = "user_removed"
+	eventActionUserChanged = "user_changed"
+)
+
+func init() {
+	mb.Registry.MustAddMetricSet(moduleName, metricsetName, New,
```

Comment:
If someone tries using this on Windows ideally they would get an error at startup or on `test config`. In order to make that happen you need to do one of these

- register a different factory function for Windows just returns a friendly error (use build tags to do this conditionally)
- not register the metricset on windows (gives a vague error)
- check for windows in this metricset factory function (`New`) and return an error

Reply:
Makes sense - I've added a check in `New`.